src/quacc/schemas/ase.py

```diff
@@ -349,6 +350,113 @@ def summarize_vib_and_thermo(
     )
 
 
+def summarize_neb_run(
+    dyn: Optimizer,
+    trajectory: Trajectory | list[Atoms] | None = None,
+    charge_and_multiplicity: tuple[int, int] | None = None,
+    additional_fields: dict[str, Any] | None = None,
+    store: Store | None | DefaultSetting = QuaccDefault,
+) -> OptSchema:
+    """
+    Summarize the NEB run results and store them in a database-friendly format.
+
+    Parameters
+    ----------
+    dyn
+        ASE Optimizer object used for the NEB run.
+    trajectory
+        Trajectory of the NEB run, either as a Trajectory object or a list of Atoms objects.
+    charge_and_multiplicity
+        Charge and spin multiplicity of the Atoms object.
+    additional_fields
+        Additional fields to add to the task document.
+    store
+        Maggma Store object to store the results in. Defaults to `QuaccSettings.STORE`.
+
+    Returns
+    -------
+    OptSchema
+        A dictionary containing the summarized NEB run results.
+    """
+    settings = get_settings()
+    store = settings.STORE if store == QuaccDefault else store
+    additional_fields = additional_fields or {}
+
+    # Get trajectory
+    if not trajectory:
+        trajectory = read(dyn.trajectory.filename, index=":")
```

Comment:
To prevent overwriting of variables, can you make this

```python
if trajectory:
    atoms_trajectory = trajectory
else:
    atoms_trajectory =  read(dyn.trajectory.filename, index=":")  # type: ignore[union-attr]
```